include/lbann/layers/learning/base_convolution.hpp

```diff
@@ -37,6 +37,8 @@
 #include "lbann/utils/random.hpp"
 #include "lbann/utils/timer.hpp"
 #include "lbann/utils/im2col.hpp"
+#include "lbann/training_algorithms/training_algorithm.hpp"
+#include "lbann/training_algorithms/sgd_training_algorithm.hpp"
```

Comment:
I don't see the training algorithms used in this file, just execution contexts. Perhaps just `#include` the appropriate `execution_context` headers.